File: 代码随想录训练营/day1(2).cpp
//
// Created by StarkLu on 2023/5/24.
//
#include "iostream"
#include "vector"
using namespace std;
class Solution {
public:
    int removeElement(vector<int>& nums, int val)
    {
        //{1,2,2,4}
        int slow = 0;

        for(int fast = 0; fast < nums.size(); fast++)
        {
            if(nums[fast] != val)
            {
                nums[slow] = nums[fast];
                slow++;
            }
        }
        return slow;
    }

};


int main()
{
    vector<int> nums = {0,1,2,2,3,0,4,2};
    int val =2;
    Solution s;
    cout << s.removeElement(nums, val) << endl;
}

File: 代码随想录训练营/day1(2).py
def removeElements(nums, val):
    """
    203. 移除链表元素
    https://leetcode-cn.com/problems/remove-linked-list-elements/
    """
    slow = 0
    fast = 0
    n  = len(nums)
    for fast in range(n):
        if nums[fast] != val:
            nums[slow] = nums[fast]
            slow += 1
    return slow
nums = [3,2,2,3]
val = 3
print(removeElements(nums, val))